homeassistant/components/leneda/config_flow.py

```diff
@@ -0,0 +1,678 @@
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self._existing_entries: list[dict[str, Any]] = []
+        self._use_existing: bool = False
+        self._selected_energy_id: str = ""
+        self._api_token: str = ""
+        self._metering_points: list[str] = []
+        self._current_metering_point: str = ""
+        self._selected_sensors: dict[str, list[str]] = {}
+        self._probing_task: Task | None = None
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: config_entries.ConfigEntry,
+    ) -> LenedaOptionsFlow:
+        """Get the options flow for this handler."""
+        return LenedaOptionsFlow(config_entry)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        # Get existing entries
+        self._existing_entries = []
+        for entry in self.hass.config_entries.async_entries(DOMAIN):
+            self._existing_entries.append(
+                {
+                    "entry_id": entry.entry_id,
+                    "energy_id": entry.data.get(CONF_ENERGY_ID, ""),
+                    "api_token": entry.data.get(CONF_API_TOKEN, ""),
+                    "title": entry.title,
+                }
+            )
+
+        # If there are no existing entries, go directly to new credentials
+        if not self._existing_entries:
+            return await self.async_step_new_credentials()
+
+        # Show menu with options
+        return self.async_show_menu(
+            step_id="user",
+            menu_options=["new_credentials", "select_existing"],
+        )
+
+    async def async_step_select_existing(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle selecting an existing configuration."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            # Mark that we are adding to an existing entry
+            self._use_existing = True
+            self._selected_energy_id = user_input["existing_energy_id"]
+            # Get the API token from the existing entry
+            for entry in self._existing_entries:
+                if entry["energy_id"] == self._selected_energy_id:
+                    self._api_token = entry["api_token"]
+                    break
+            return await self.async_step_add_metering_point()
+
+        # Create options for existing energy IDs
+        energy_id_options = {
+            entry["energy_id"]: f"{entry['energy_id']}"
+            for entry in self._existing_entries
+        }
+
+        return self.async_show_form(
+            step_id="select_existing",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(
+                        "existing_energy_id",
+                        description="config.step.select_existing.data.existing_energy_id",
```

Comment:
I don't think the description is required here because Home Assistant should pick that up from generated translations, or strings.json, under the `data_description` key (I think?)

Reply:
Yes absolutely, I have double checked and removed the redundant descriptions (and also cleaned up strings.json)